src/Router/Middleware/NotFound.php

```diff
@@ -0,0 +1,105 @@
+<?php
+
+namespace Jasny\Router\Middleware;
+
+use Jasny\Router\Routes\Glob;
+use Psr\Http\Message\ServerRequestInterface;
+use Psr\Http\Message\ResponseInterface;
+
+/**
+ * Set response to 'not found' or 'method not allowed' if route is non exist 
+ */
+class NotFound
+{    
+    /**
+     * Routes
+     * @var array
+     */
+    protected $routes = [];
+
+    /**
+     * Action for 'not found' case
+     * @var callback|int
+     **/
+    protected $notFound = null;
+
+    /**
+     * Action for 'method not allowed' case
+     * @var string
+     **/
+    protected $methodNotAllowed = null;
+
+    /**
+     * Class constructor
+     * 
+     * @param array $routes
+     * @param callback|int $notFound 
+     * @param callback|int $methodNotAllowed 
+     */
+    public function __construct(array $routes, $notFound = 404, $methodNotAllowed = null)
+    {
+        if (!in_array($notFound, [404, '404'], true) && !is_callable($notFound)) {
+            throw new \InvalidArgumentException("'Not found' parameter should be '404' or a callback");
+        }
+
+        if ($methodNotAllowed && !in_array($methodNotAllowed, [405, '405'], true) && !is_callable($methodNotAllowed)) {
```

Comment:
You should be able to use any valid http status code (100 - 999) for `$methodNotAllowed`.
You need to do `isset($methodNotAllowed)`, so `0` and `false` will also cause an exception.